examples/rtl_433_mqtt_hass.py

```diff
@@ -1053,13 +1053,67 @@ def run():
     if not args.password and 'MQTT_PASSWORD' in os.environ:
         args.password = os.environ['MQTT_PASSWORD']
 
-    if not args.user or not args.password:
-        logging.warning("User or password is not set. Check credentials if subscriptions do not return messages.")
-
     if args.ids:
         ids = ', '.join(str(id) for id in args.ids)
         logging.info("Only discovering devices with ids: [%s]" % ids)
     else:
         logging.info("Discovering all devices")
 
+    found_config = False 
+    try:
+        with open(find_file_in_path(CONFIG_FILENAME, CONFIG_PATHS)) as f:
+            found = re.findall('^output (.*)$', f.read(), re.MULTILINE)
+            devices = None
+            events = None
+            retain = None
+            user = None 
+            password = None 
+            host = None 
+            port = None
+            for find in found: 
+                if "mqtt" not in find: continue
+                host = re.findall(r'mqtt://(.*?),', find)[0]
+                devices = re.findall(r'devices=(.*?)(?:,+|$)', find)
+                if len(devices) > 0: devices = devices[0]
+                events = re.findall(r'events=(.*?)(?:,+|$)', find)
+                if len(events) > 0: events = events[0]
+                retain = re.findall(r'retain=(.*?)(?:,+|$)', find)
+                if len(retain) > 0: retain = retain[0]
+
+                if '@' in host:
+                    user, password = host.split('@')[0].split(':')
+                    host = host.split('@')[1]
+                if ':' in host:
+                    host, port = host.split(':')
+                found_config = True
+                break 
+    except FileNotFoundError:
+        logging.warning("Config file not found")
```

Comment:
It would be helpful to be less ambiguous about what config file you are talking about in this and the other log messages. 

I would explicitly mention that it is rtl_433's config file and/or the path.

Some of the Home Assistant add-ons have their own config files for this script.